### PR TITLE
Deserialize Single Element

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
Refactor XML and TSV deserializer to allow single element deserialization

### CODE REVIEW

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt [hunk 1]
```diff
@@ -38,6 +39,15 @@ class TsvDeserializer(private val chunkProcessor: ChunkProcessor = ChunkProcesso
         return context.getSubmission()
     }
 
+    fun <T> deserializeElement(pageTab: String, type: Class<out T>): T {
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt]: 👍

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/ChunkProcessor.kt [hunk 2]
```diff
@@ -46,6 +47,20 @@ class ChunkProcessor {
             attributes = toAttributes(tsvChunk.lines))
     }
 
+    fun processIsolatedChunk(chunk: TsvChunk): Any {
+        val element: Any
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/ChunkProcessor.kt]: 👍

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/TsvSerializer.kt [hunk 3]
```diff
@@ -12,6 +12,8 @@ class TsvSerializer(
 
     fun <T> serialize(element: T) = tsvSerializer.serialize(element)
 
+    fun <T> deserializeElement(pageTab: String, type: Class<out T>) = tsvDeserializer.deserializeElement(pageTab, type)
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/TsvSerializer.kt]: 👍

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt [hunk 4]
```diff
@@ -38,6 +39,15 @@ class TsvDeserializer(private val chunkProcessor: ChunkProcessor = ChunkProcesso
...
+        val chunks: MutableList<TsvChunk> = chunkerize(pageTab)
+        chunks.ifEmpty { throw InvalidElementException(EMPTY_ELEMENT_ERROR_MSG) }
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/TsvDeserializer.kt]: Agreed

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/ChunkProcessor.kt [hunk 5]
```diff
@@ -46,6 +47,20 @@ class ChunkProcessor {
...
+
+        when (chunk) {
+            is LinkChunk -> element = chunk.asLink()
+            is FileChunk -> element = chunk.asFile()
+            is LinksTableChunk -> element = chunk.asTable()
+            is FileTableChunk -> element = chunk.asTable()
+            else -> throw InvalidChunkException(chunk)
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/ChunkProcessor.kt]: Done

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/xml/XmlSerializer.kt [hunk 6]
```diff
@@ -22,31 +24,46 @@ import com.fasterxml.jackson.dataformat.xml.XmlMapper
...
 
-    fun <T> serialize(element: T): String {
-        return mapper.writeValueAsString(element)
-    }
+    fun deserialize(value: String) = deserialize(value, Submission::class.java)
+
+    fun <T> deserialize(element: String, type: Class<out T>): T {
+        var deserialized: Any? = null
+        val xml = buildXmlFile(element)
+        val attributeDeserializer = AttributeXmlDeserializer(DetailsXmlDeserializer())
```
Comment (Juan-EBI) [inline commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/xml/XmlSerializer.kt]: I believe this is not necessary at all, (check json example) jackson object mapper is able to identify which deserializer call based on the class type

**Pull request merged**